runtime/op/meta/sequence.go

```diff
@@ -147,6 +148,12 @@ func newObjectScanner(ctx context.Context, zctx *zed.Context, pool *lake.Pool, o
 	}
 	rc, err := object.NewReader(ctx, pool.Storage(), pool.DataPath, ranges)
 	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			// If the object does not exist return a zbuf.Puller that returns a
+			// single value- a zed error.
+			vals := zbuf.NewArray([]zed.Value{*zctx.NewError(err)})
+			return zbuf.NewPuller(vals), nil
+		}
```

Comment:
This is a neat idea and we should eventually add this sort of error tolerance as option, but I don't think it's the right default behavior.

Reply:
This was part of the original issue description https://github.com/brimdata/zed/issues/4113:

> In this case, perhaps we should emit a structured Zed error describing the meta-data of the object that was unavailable.

Are you saying that the query should just return an error?

Reply:
Yes, it should return an error.

Reply:
ok